src/err.rs

```diff
@@ -450,14 +472,25 @@ impl<'a> IntoPy<PyObject> for &'a PyErr {
 
 /// Convert `PyDowncastError` to Python `TypeError`.
 impl std::convert::From<PyDowncastError> for PyErr {
-    fn from(_err: PyDowncastError) -> PyErr {
-        exceptions::TypeError.into()
+    fn from(err: PyDowncastError) -> PyErr {
+        match (err.from, err.to) {
+            (Some(from), Some(to)) => {
+                exceptions::TypeError::py_err(format!("Failed to convert from {} to {}", from, to))
+            }
+            (Some(from), None) => {
+                exceptions::TypeError::py_err(format!("Failed to convert from {}", from))
+            }
+            (None, Some(to)) => {
+                exceptions::TypeError::py_err(format!("Failed to convert to {}", to))
+            }
+            (None, None) => exceptions::TypeError::py_err("Failed to convert"),
+        }
```

Comment:
If you change the `PyDowncastError` struct to my suggested form (esp. without `Option`), and change `Debug` impl to `Display` as suggested, then I think that this can just re-use the `Display` implementation:

```suggestion
        exceptions::TypeError::py_err(e.to_string())
```